pytest_plugins/logging_hooks.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+import logzero
+import pytest
+from xdist import is_xdist_worker
+
+from robottelo.logging import DEFAULT_DATE_FORMAT
+from robottelo.logging import logger
+from robottelo.logging import robottelo_log_dir
+from robottelo.logging import robottelo_log_file
+
+try:
+    from pytest_reportportal import RPLogger
+    from pytest_reportportal import RPLogHandler
+except ImportError:
+    pass
+
+
+@pytest.fixture(autouse=True, scope='session')
+def configure_logging(request, worker_id):
+    """Handle xdist and ReportPortal logging configuration at session start
+
+    Set up a separate logger for each pytest-xdist worker
+    if worker_id != 'master' then xdist is running in multi-threading so
+    a logfile named 'robottelo_gw{worker_id}.log' will be created.
+
+    Add a handler for ReportPortal logging
+    """
+    worker_formatter = logzero.LogFormatter(
+        fmt=f'%(asctime)s - {worker_id} - %(name)s - %(levelname)s - %(message)s',
+        datefmt=DEFAULT_DATE_FORMAT,
+    )
+    use_rp_logger = hasattr(request.node.config, 'py_test_service')
+    if use_rp_logger:
+        logging.setLoggerClass(RPLogger)
+
+    if is_xdist_worker(request):
+        if f'{worker_id}' not in [h.get_name() for h in logger.handlers]:
+            # Track the core logger's file handler level, set it in case core logger wasn't set
+            worker_log_level = 'INFO'
+            handlers_to_remove = [
+                h
+                for h in logger.handlers
+                if isinstance(h, logging.FileHandler)
+                and getattr(h, 'baseFilename', None) == str(robottelo_log_file)
+            ]
+            for handler in handlers_to_remove:
+                logger.removeHandler(handler)
+                worker_log_level = handler.level
+            worker_handler = logging.FileHandler(
+                robottelo_log_dir.joinpath(f'robottelo_{worker_id}.log')
+            )
+            worker_handler.set_name(f'{worker_id}')
+            worker_handler.setFormatter(worker_formatter)
+            worker_handler.setLevel(worker_log_level)
+            logger.addHandler(worker_handler)
+
+            if use_rp_logger:
+                rp_handler = RPLogHandler(request.node.config.py_test_service)
+                rp_handler.setFormatter(worker_formatter)
+                # logger.addHandler(rp_handler)
+
+
+def pytest_runtest_logstart(nodeid, location):
```

Comment:
The only issue I see with the logging behavior in this PR are these two hooks.

When running in xdist, the messages get written to the `robottelo_gw#.log` file - but they also get written to `robottelo.log`, resulting in robottelo.log just having all of the start/finish test lines with no other content when running in xdist.

The logger at this scope should be the xdist worker logger, but for some reason it also writes to the original file handler that was removed in the logging configuration hook/fixture.  I suspect its pytest capturing the output from these fixtures specifically?

I'm not too concerned about the behavior, but it's not as clean as it could be.

Reply:
i'm fine with it logging to the main file as well, could be better for people who want to see everything in one place.

Reply:
Well - it _only_ logs the messages from these hooks. _Everything_ else only goes to the xdist worker log file.